PluralKit.Bot/Commands/Member.cs

```diff
@@ -70,21 +75,101 @@ public Member(EmbedService embeds, IDatabase db, ModelRepository repo, HttpClien
             if (memberCount >= memberLimit)
-                await ctx.Reply($"{Emojis.Warn} You have reached the per-system member limit ({memberLimit}). You will be unable to create additional members until existing members are deleted.");
+                text += $"\n{Emojis.Warn} You have reached the per-system member limit ({memberLimit}). You will be unable to create additional members until existing members are deleted.";
             else if (memberCount >= Limits.MaxMembersWarnThreshold(memberLimit))
-                await ctx.Reply($"{Emojis.Warn} You are approaching the per-system member limit ({memberCount} / {memberLimit} members). Please review your member list for unused or duplicate members.");
+                text += $"\n{Emojis.Warn} You are approaching the per-system member limit ({memberCount} / {memberLimit} members). Please review your member list for unused or duplicate members.";
+
+            Embed? embed = null;
+            if (ctx.System.WelcomeMessage != null) {
+                var eb = new EmbedBuilder()
+                    .Title($"Welcome, {memberName}!")
+                    .Color(ctx.System.Color?.ToDiscordColor() ?? DiscordUtils.Gray)
+                    .Description(ctx.System.WelcomeMessage)
+                    .Footer(new($"System ID: {ctx.System.Hid}"));
+
+                switch (ctx.System.WelcomeMessageMode) {
+                    case WelcomeMessageMode.Off: break;
+                    case WelcomeMessageMode.Inline:
+                    {
+                        embed = eb.Build();
+                        break;
+                    }
+                    case WelcomeMessageMode.DM:
+                    {
+                        try {
+                            var dm = await ctx.Cache.GetOrCreateDmChannel(ctx.Rest, ctx.Author.Id);
+                            await ctx.Rest.CreateMessage(dm.Id, new MessageRequest{
+                                Embed = eb.Build()
+                            });
+                        }
+                        catch (ForbiddenException)
+                        {
+                            if (ctx.Channel.Type != Channel.ChannelType.Dm)
+                                text += $"\n{Emojis.Error} Could not send welcome message in DMs. Are your DMs closed?";
+                        }
+                        break;
+                    }
+                    case WelcomeMessageMode.CustomChannel:
+                    {
+                        // This shouldn't ever happen, but if it does, act as if there's no message set at all.
+                        if (ctx.System.WelcomeMessageChannel == null) break;
+
+                        try {
+                            var channel = await ctx.Cache.GetOrFetchChannel(ctx.Rest, (ulong)ctx.System.WelcomeMessageChannel);
```

Comment:
if I don't cast it to a ulong, the compiler complains that it can't convert `ulong?` to `ulong`--but it'll never be null as that's checked just a few lines earlier, I just couldn't figure out a cleaner way to do it

Reply:
oh. use `.Value`